### DISCUSSION TITLE
SQLiteStudio (3.3.3) won't let me import CSV file

### DISCUSSION THREAD

**Discussion by savannahkeiffer** [pawelsalawa/sqlitestudio#4104] [Q&A]
I downloaded SQLiteStudio last week in the hopes of using it to make my job easier. I'm an invoicing analyst, so my job consists of generating invoices all day in our UI, most of which include CSV or TXT files we manually have to balance. However, I believe I can make this step in the process easier by using SQL to extract the necessary information from these files rather than doing it manually. 

**Here's where my issue comes in:**

I've successfully created databases and imported CSV files into these databases and even wrote the code to generate the results I need. Everything was working fine until Sunday. Suddenly, I tried opening SQLiteStudio so I could work on more code for other customers and I couldn't import anything. Every time I go to do so, the "Finish" button at the bottom of the screen is greyed out.

**Here's what I've tried:**

I've checked to make sure all required fields were filled out, I've tried restarting the desktop app and reopening, I've tried restarting my computer and uninstalling the app and reinstalling it, none of which has helped. I've made sure that all databases and files are in the same folders as well. 

I've attached a screenshot to show you what I'm talking about. I'm having no problems creating or connecting to a database or running code to analyze any of the data I've already imported. It's only the import feature that seems to be the problem. My computer runs on Windows 10 version 20H2.

![image](https://user-images.githubusercontent.com/50883745/117881233-3f19dd00-b266-11eb-9f79-ec22d9049a98.png)

**Comment by pawelsalawa** [comment]
To be honest I don't know why is it like that. I cannot reproduce it. If you move one step back (using arrow in top-left corner) what is the status there?

**Reply by savannahkeiffer** [reply]
So I think I found out what happened! I was trying to import a txt file, but I couldn't figure out the regular expression required to open it properly, so it kept erroring out. I decided to move on to another CSV file I wanted to try in the meantime, but every time I tried to import a CSV, the fields I have screenshotted above were already generated with all the txt file info (i.e. "CSV" at the top was "RegEx", input file was the txt file I was originally looking at, etc). Basically, I tried importing the original txt file with a very basic regular expression just to get past that screen. Once the data loaded (incorrectly of course since I used a random, basic regex), I tried importing the CSV again and it worked! It's like SQLiteStudio wouldn't let me import another file until I finished trying to import the txt file that kept erroring out?

**Reply by pawelsalawa** [reply]
It should let you.

I found it - when you switch to RegExp, leave expression incorrect (with red error indicator) and switch back to CSV format - your problem will appear. It's a matter of incorrect state refreshing. I'll fix it.

**Reply by pawelsalawa** [reply]
Reported #4105
